src/job/jobHandler.ts

```diff
@@ -256,25 +254,20 @@ export abstract class JobHandler {
 
   @throwIfJobInterupted()
   private async prepNextGenBuild(): Promise<void> {
-    if (this.isbuildNextGen()) {
-      await this._validator.throwIfBranchNotConfigured(this.currJob);
-      await this.constructPrefix();
-      // TODO: Look into moving the generation of manifestPrefix into the manifestJobHandler,
-      // as well as reducing difficult-to-debug state changes
-      // if this payload is NOT aliased or if it's the primary alias, we need the index path
-      if (!this.currJob.payload.aliased || (this.currJob.payload.aliased && this.currJob.payload.primaryAlias)) {
-        this.currJob.payload.manifestPrefix = this.constructManifestPrefix();
-        this._logger.info(this.currJob._id, `Created payload manifestPrefix: ${this.currJob.payload.manifestPrefix}`);
-      }
+    await this._validator.throwIfBranchNotConfigured(this.currJob);
+    await this.constructPrefix();
+    // TODO: Look into moving the generation of manifestPrefix into the manifestJobHandler,
+    // as well as reducing difficult-to-debug state changes
+    // if this payload is NOT aliased or if it's the primary alias, we need the index path
+    if (!this.currJob.payload.aliased || (this.currJob.payload.aliased && this.currJob.payload.primaryAlias)) {
+      this.currJob.payload.manifestPrefix = this.constructManifestPrefix();
+      this._logger.info(this.currJob._id, `Created payload manifestPrefix: ${this.currJob.payload.manifestPrefix}`);
+    }
 
-      this.prepStageSpecificNextGenCommands();
-      this.constructEnvVars();
-      this.currJob.payload.isNextGen = true;
-      if (this._currJob.payload.jobType === 'productionDeploy') {
-        this._validator.throwIfNotPublishable(this._currJob);
-      }
-    } else {
-      this.currJob.payload.isNextGen = false;
+    this.prepStageSpecificNextGenCommands();
```

Comment:
We should update the `prepStageSpecificNextGenCommands();`. That method contains stuff that we can clean up/simplify. More specifically, that function depends on mutations made to the `this.currJob.buildCommands` property in the `this.prepBuildCommands` method.

Anywhere we see `. /venv/bin/activate` or `make (parse/html/stage/deploy)` we should remove that.

Reply:
should `make (parse/html/stage/deploy)` be replaced with something else?

Reply:
Good question. They should be replaced with the next-gen equivalent  e.g. make parse should be replaced with next-gen-parse. `make publish` should be removed as well, good catch!